tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -1669,55 +1669,55 @@ public function dataBinaryOperations(): array
 				'__DIR__',
 			],
 			[
-				'int', // if the only argument in min is array, lowest value in that array is returned
+				'int(1)|int(2)|int(3)', // if the only argument in min is array, lowest value in that array is returned
 				'min([1, 2, 3])',
 			],
 			[
-				'array<int, int>',
+				'array<int(0)|int(1)|int(2), int(1)|int(2)|int(3)|int(4)|int(5)>',
```

Comment:
Just an idea - what if constant scalar types' description consisted only of the value? The type could be omitted... So this would look like `array<0|1|2>`...

Reply:
I'd also add a setting because having the specific values in descriptions can get really messy in case of huge arrays. I'd turn the constant values `on` for the needs of PHPStan's tests but the default for the user would be `off`.

Reply:
It's not an easy problem to solve. Without the constant value in the description, you may encounter error messages which are confusing – this is already the case, because we currently show constant values of only integers and floats. See https://phpstan.org/r/1ae4341094557cf1a052505e7b381218 for some examples of situations in which current solutions work well (first 2), and of situations in which current solution falls short (last 2).

So on one hand there situations in which we would like to show less information (because it's good enough and easier to read), and on the other hand there are situations in which we would like to show even more information than now (i.e. constant string and constant array values). Maybe specific rules such as StrictComparisonOfDifferentTypesRule could override the default *verbosity setting* and always show full value.

Reply:
So maybe we could add an optional parameter for the `Type::describe()` method? Can be added in the next phase, not in this PR, but I'd like to keep type descriptions as simple as possible, they are already quite complex since the intersection types...